R/network.R

```diff
@@ -13,57 +13,49 @@
 #'
 #' @param project_dependencies A parsed depends project by \code{\link{parse_dependencies}}.
 #' @param sdsmj_path the path to save the structural dsm (*-sdsm.json).
-#' @param is_sorted whether to sort the variables (filenames) in the sdsm.json file.
+#' @param is_sorted whether to sort the variables (filenames) in the sdsm.json file (optional).
 #' @export
 #' @family edgelists
 #' @family dv8
-#' @seealso \code{\link{parse_dependencies}} to get a table of parsed dependencies needed
-#'  as input into \code{\link{transform_dependencies_to_sdsmj}},
-#'  \code{\link{transform_gitlog_to_hdsmj}} to perform a similar transformation into a *-dsm.json using a gitlog,
-#' \code{\link{graph_to_dsmj}} to generate a *-dsm.json file. A *-sdsm.json will be returned for \code{\link{transform_dependencies_to_sdsmj}}
-#' and a *-hdsm.json will be returned for \code{\link{transform_gitlog_to_hdsmj}}
+#' @seealso \code{\link{parse_dependencies}} to get a table of parsed dependencies needed as input into \code{\link{transform_dependencies_to_sdsmj}},
+#' \code{\link{transform_gitlog_to_hdsmj}} to perform a similar transformation into a *-dsm.json using a gitlog,
+#' \code{\link{transform_temporal_gitlog_to_adsmj}} to perform a similar transformation into a *-dsm.json using a temporal gitlog,
+#' \code{\link{graph_to_dsmj}} to generate a *-dsm.json file.
 transform_dependencies_to_sdsmj <- function(project_dependencies, sdsmj_path, is_sorted=FALSE){
   # Convert table to long form
   project_dependencies[["edgelist"]] <- melt(project_dependencies[["edgelist"]],id.vars <- c("src_filepath","dest_filepath"), variable.name = "label")
 
-  project_dependencies[["nodes"]] <- setnames(x=project_dependencies[["nodes"]],
-                                                 old = c("filepath"),
-                                                 new = c("name"))
+  setnames(x=project_dependencies[["nodes"]], old = c("filepath"), new = c("name"))
```

Comment:
For some reason, once I run transform_dependencies_to_sdsmj() on my parsed dependencies_table, if I try to load my parsed_dependencies_table again then parsed_dependencies_table[["nodes"]] column called "filepath" is renamed to "name", which I did in transform_dependencies_to_sdsmj(). However, in parsed_dependencies_table[["edgelist"]], the column names are not renamed. I don't understand why this is.

Reply:
I'm confused about what the issue is here. In the code above you are only modifying the [["nodes"]] table, so the [["edgelist"]] columns remain unaffected. Regardless, the edgelist column names should be `from` and `to` for the filepaths, not name. Did I misunderstand the question?

Reply:
Sorry, the line below this shows where I renamed the edgelist columns.

```r
  setnames(x=project_dependencies[["edgelist"]], old = c("src_filepath","dest_filepath", "value"),
           new = c("from","to", "weight"))
```

And in regards to the nodes table, I renamed it to "name" from "filepath" since that was mentioned in my dv8 functions issue a while back https://github.com/sailuh/kaiaulu/issues/168#issuecomment-1509333620.

So to clarify the problem, I have this table
```r
parsed_depends_table <- kaiaulu::parse_dependencies(depends_path, git_repo_path, lang, output_dir)
parsed_depends_table
```

and then I call my function
```r
kaiaulu::transform_dependencies_to_sdsmj(parsed_depends_table,sdsmj_path="calc-april29-depends.json",is_sorted=TRUE)
```

and then I try to print the parsed_depends_table again
The nodes table "filepath" column is now "name"
<img width="655" alt="Screenshot 2023-04-29 at 7 55 04 PM" src="https://user-images.githubusercontent.com/89750594/235338130-1ada854d-2d12-44fe-94cc-c615f46b5aba.png">

while the edgelist table columns did not change
<img width="512" alt="Screenshot 2023-04-29 at 7 55 10 PM" src="https://user-images.githubusercontent.com/89750594/235338149-0137b4c2-b5d8-45a0-9c7f-be832a32ede9.png">

I just think this behavior is odd. Also, should the function even be modifying the original "parsed_depends_table" columns? If the user wanted to use that parsed_depends_table for something else, I think this would cause issues.

Reply:
Your concern is valid.

If the issue is, it is modifying the original table, try looking up Google how data.table modifies objects by reference instead of copying. This is what makes data.table so efficient, aside from being implemented in C underneath. 

On your transform function, you want to copy the object and apply setnames to the copied object instead of setnames to the original. Accordingly, you also want to pass the copied object to graph_to_dsmj() too.

Have a look on this: 

https://rdrr.io/cran/data.table/man/copy.html#:~:text=data.table%20provides%20functions%20that%20operate%20on%20objects%20by,used%20to%20subassign%20to%20a%20column%20by%20reference.

Let me know if I misunderstood or it is not addressing your issue.

Reply:
Thank you, this works!